### PR TITLE
Simplify constant switch first

### PR DESCRIPTION
**Pull request by chriseth**
Simplify switch with constant expression first because it allows the biggest simplification.

Depends on #6207

### CODE REVIEW

File: libyul/optimiser/StructuralSimplifier.cpp [hunk 1]
```diff
@@ -202,10 +205,11 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 		_statements,
 		[&](Statement& _stmt) -> OptionalStatements
 		{
-			visit(_stmt);
 			OptionalStatements result = boost::apply_visitor(visitor, _stmt);
 			if (result)
 				simplify(*result);
+			else
+				visit(_stmt);
```
Comment (ekpyron) [inline libyul/optimiser/StructuralSimplifier.cpp]: Should we put this in its own PR and give it a bugfix changelog entry? I mean: it could even incorrectly change the semantics of code before... Actually given #5959 I'd even consider it for the buglist...

Comment (ekpyron) [reply libyul/optimiser/StructuralSimplifier.cpp]: Ah nevermind, it has at least a Changelog entry here, I missed that.

Comment (chriseth) [reply libyul/optimiser/StructuralSimplifier.cpp]: -> #6207

Comment (ekpyron) [reply libyul/optimiser/StructuralSimplifier.cpp]: Ah :-). I should read PR descriptions :-).

**Review by ekpyron — DISMISSED** [review]
